### PR TITLE
Vih 3839 polly on deleting users

### PR DESCRIPTION
**Pull request by ddikman**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-3839

### Change description ###
Added in polly to retry on user deletion. It's needed for next step when user creation will complete before the user is fully instantiated.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs [hunk 1]
```diff
@@ -39,14 +39,26 @@ public static bool RemoveTheUserFromTheGroup(string user, string groupId, string
 
         public static bool DeleteTheUserFromAd(string user, string token)
         {
-            using (var client = new HttpClient())
+            
+            var tenantId = TestConfig.Instance.AzureAd.TenantId;
+            var policy = Policy.HandleResult<HttpResponseMessage>(r => r.StatusCode == HttpStatusCode.NotFound)
```
Comment (rehmanab) [inline UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs:44]: Do we want to also handle for other status codes?

File: UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs [hunk 2]
```diff
@@ -39,14 +39,26 @@ public static bool RemoveTheUserFromTheGroup(string user, string groupId, string
 
         public static bool DeleteTheUserFromAd(string user, string token)
```
Comment (rehmanab) [inline UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs]: async ?

File: UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs (hunk 1 above)
Comment (ddikman) [reply UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs:44]: I wouldn't retry on other status code, if it's BadRequest or something like that, I want it to simply bubble the error (I return it with the IsSuccessResponseeCode below).

Admittedly this code is far from perfect though, I'll try to improve it in upcoming PR.

File: UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs (hunk 2 above)
Comment (ddikman) [reply UserApi/Testing.Common/ActiveDirectory/ActiveDirectoryUser.cs]: When I first wrote it I wanted to keep the changes to a minimum but as it's just a few changes in this PR making it async in the same go makes sense. Changed it.

**Review by rehmanab — APPROVED** [review]

**Pull request merged**